src/components/application_manager/include/application_manager/hmi_capabilities_impl.h

```diff
@@ -90,491 +76,338 @@ class HMICapabilitiesImpl : public HMICapabilities {
+      const OVERRIDE;
+
+  void UpdateRequestsRequiredForCapabilities(
+      hmi_apis::FunctionID::eType requested_interface) OVERRIDE;
+
+  bool MatchesCCPUVersion(const std::string& ccpu_version) const OVERRIDE;
+
+  void OnSoftwareVersionReceived(const std::string& ccpu_version) OVERRIDE;
+
+  void UpdateCachedCapabilities() OVERRIDE;
+
  protected:
-  /*
+  /**
    * @brief Loads capabilities from local file in case SDL was launched
    * without HMI
    *
    * @return TRUE if capabilities loaded successfully, otherwise FALSE.
    */
-  bool load_capabilities_from_file();
-
-  /*
-   * @brief function checks if json member exists
-   *
-   * @param json_member from file hmi_capabilities.json
-   * @param name_of_member name which we should check
-   *     hmi_capabilities.json
-   *
-   * @returns TRUE if member exists and returns FALSE if
-   * member does not exist.
-   */
-  bool check_existing_json_member(const Json::Value& json_member,
-                                  const char* name_of_member) const OVERRIDE;
+  bool LoadCapabilitiesFromFile();
 
-  /*
+  /**
    * @brief function converts json object "languages" to smart object
-   *
    * @param json_languages from file hmi_capabilities.json
    * @param languages - the converted object
    *
    */
   void convert_json_languages_to_obj(
       const Json::Value& json_languages,
-      smart_objects::SmartObject& languages) const OVERRIDE;
+      smart_objects::SmartObject& languages) const;
 
-  /*
+  /**
    * @brief function that converts a single entry of audio pass thru capability
    *        to smart object
-   *
    * @param capability json object that represents a single entry of audio pass
    *        thru capability
    * @param output_so the converted object
    */
   void convert_audio_capability_to_obj(
       const Json::Value& capability,
-      smart_objects::SmartObject& output_so) const OVERRIDE;
+      smart_objects::SmartObject& output_so) const;
 
  private:
+  /**
+   * @brief Checks are all updating fields are currently saved in the JSON
+   * structure
+   * @param root_node reference to root node of JSON structure
+   * @param interface_name name of interface to check
+   * @param sections_to_check reference to list of fields to check
+   * @return true if all fields from the list are saved in the JSON structure,
+   * otherwise returns false
+   */
+  bool AllFieldsSaved(const Json::Value& root_node,
+                      const std::string& interface_name,
+                      const std::vector<std::string>& sections_to_check) const;
+
+  /**
+   * @brief Remove received interface from default initialized capabilities
+   * @param requested_interface interface which should be removed
+   */
+  void RemoveFromRequestsRequiredForCapabilities(
+      hmi_apis::FunctionID::eType requested_interface);
+
+  /**
+   * @brief Setting HMICooperating to true for respond all holding RAI requests
```

Comment:
Setting HMICooperating to true to respond to all holding RAI requests

Reply:
@ydementieiev 
- Description: 
Checks if there are pending requests for capabilities and sets HMICooperating to 
"true" if no pending requests for capabilities present, otherwise does not change HMICooperating. 
- The function performs several actions. I would recommend to split it to 2 functions: 
   - The first one checks if pending requests for caps are present
   - The second one changes HMICooperating value

Reply:
@LitvinenkoIra @AByzhynar for more readable I removed this function and used simple `if`  instead [here](https://github.com/LuxoftSDL/sdl_core/pull/35/commits/6112031da80fbfeac110ab8f2570b31e61f394aa)

Reply:
@ydementieiev :+1: